### PR TITLE
Introduce formatter black

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Reformat code using black
- Travis will fail from now on if code was not formatted with black

closes # https://github.com/RasaHQ/rasa_nlu/issues/3205

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by tmbo** [comment]
As discussed, please 
- add black to dev requirements
- add instructions on running black to the pull request github template (e.g. to the checklist?)

**Comment by tabergma** [comment]
Unfortunately, we cannot add `black` to the requirements as `black` requires >= python3.6 and travis also builds with python3.5.

**Comment by tabergma** [comment]
@tmbo Do you have any other comments? Would like to merge :)

File: .travis.yml [hunk 1]
```diff
@@ -35,6 +35,10 @@ after_success:
   - coveralls
 jobs:
   include:
+  - stage: test
+    name: "Black Formatting"
+    python: 3.6
+    script: pip install black; black --check .
```
Comment (tmbo) [inline .travis.yml]:
```suggestion
    install: pip install black
    script: black --check .
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -11,6 +11,7 @@ This project adheres to `Semantic Versioning`_ starting with version 1.0.
 
 Added
 -----
+- add formatter 'black'
```
Comment (tmbo) [inline CHANGELOG.rst]:
```suggestion
- added formatter 'black' to check and correct python code style issues
```

**Pull request merged**